bin/install.sh

```diff
@@ -261,48 +260,59 @@ function _prepare {
-  echo "# copying sources to $ASSETS_DIR/src"
-  cp -r "$WORK_DIR"/src/* "$ASSETS_DIR/src"
+  echo "# copying sources to $SOCKET_HOME/src"
+  cp -r "$CWD"/src/* "$SOCKET_HOME/src"
 
   if test -d "$BUILD_DIR/$arch-$platform/objects"; then
-    echo "# copying objects to $ASSETS_DIR/objects/$arch-$platform"
-    rm -rf "$ASSETS_DIR/objects/$arch-$platform"
-    mkdir -p "$ASSETS_DIR/objects/$arch-$platform"
-    cp -r "$BUILD_DIR/$arch-$platform/objects"/* "$ASSETS_DIR/objects/$arch-$platform"
+    echo "# copying objects to $SOCKET_HOME/objects/$arch-$platform"
+    rm -rf "$SOCKET_HOME/objects/$arch-$platform"
+    mkdir -p "$SOCKET_HOME/objects/$arch-$platform"
+    cp -r "$BUILD_DIR/$arch-$platform/objects"/* "$SOCKET_HOME/objects/$arch-$platform"
   fi
 
   if test -d "$BUILD_DIR"/lib; then
-    echo "# copying libraries to $ASSETS_DIR/lib"
-    mkdir -p "$ASSETS_DIR/lib"
-    cp -fr "$BUILD_DIR"/lib/*.a "$ASSETS_DIR/lib/"
+    echo "# copying libraries to $SOCKET_HOME/lib"
+    mkdir -p "$SOCKET_HOME/lib"
+    cp -fr "$BUILD_DIR"/lib/*.a "$SOCKET_HOME/lib/"
   fi
 
   if test -d "$BUILD_DIR/$arch-$platform"/lib; then
-    echo "# copying libraries to $ASSETS_DIR/lib/$arch-$platform"
-    rm -rf "$ASSETS_DIR/lib/$arch-$platform"
-    mkdir -p "$ASSETS_DIR/lib/$arch-$platform"
-    cp -fr "$BUILD_DIR/$arch-$platform"/lib/*.a "$ASSETS_DIR/lib/$arch-$platform"
+    echo "# copying libraries to $SOCKET_HOME/lib/$arch-$platform"
+    rm -rf "$SOCKET_HOME/lib/$arch-$platform"
+    mkdir -p "$SOCKET_HOME/lib/$arch-$platform"
+    cp -fr "$BUILD_DIR/$arch-$platform"/lib/*.a "$SOCKET_HOME/lib/$arch-$platform"
   fi
 
-  rm -rf "$ASSETS_DIR/include"
-  mkdir -p "$ASSETS_DIR/include"
-  #cp -rf "$WORK_DIR"/include/* $ASSETS_DIR/include
-  cp -rf "$BUILD_DIR"/uv/include/* $ASSETS_DIR/include
+  rm -rf "$SOCKET_HOME/include"
+  mkdir -p "$SOCKET_HOME/include"
+  #cp -rf "$CWD"/include/* $SOCKET_HOME/include
+  cp -rf "$BUILD_DIR"/uv/include/* $SOCKET_HOME/include
 }
 
 function _install_cli {
   local arch="$(uname -m)"
-  local platform="desktop"
-  if [ -z "$TEST" ]; then
-    local binDest="/usr/local/bin/ssc"
-    echo "# moving binary to $binDest (prompting to copy file into directory)"
-    sudo mkdir -p /usr/local/bin
-    sudo rm -f "$binDest"
-    sudo cp "$BUILD_DIR/$arch-$platform"/bin/ssc $binDest
-  fi
 
-  die $? "not ok - unable to move binary into place"
-  echo "ok - done. type 'ssc -h' for help"
+  if [ -z "$TEST" ] && [ -z "$NO_INSTALL" ]; then
+    echo "# moving binary to '$SOCKET_HOME/bin' (prompting to copy file into directory)"
+
+    cp -f "$BUILD_DIR/$arch-desktop"/bin/* "$SOCKET_HOME/bin"
+    die $? "not ok - unable to move binary into '$SOCKET_HOME'"
+
+    local status="$(ln -sf "$SOCKET_HOME/bin"/* "$PREFIX/bin" 2>&1)"
+    local rc=$?
+
+    if [[ " $status " =~ " Permission denied " ]]; then
+      echo "warn - Failed to link binrary to '$PREFIX/bin': Trying 'sudo'"
+      sudo ln -sf "$SOCKET_HOME/bin"/* "$PREFIX/bin"
```

Comment:
Is this necessary? I'd think the script would succeed if they had run it with correct permissions.

Reply:
yeah i'll come back to this one!